Sources/FluentKit/Properties/Child.swift

```diff
@@ -0,0 +1,226 @@
+        self.parentKey = .optional(optionalParent)
+    }
+
+    public var wrappedValue: To {
+        get {
+            guard let value = self.value else {
+                fatalError("Child relation not eager loaded, use $ prefix to access: \(name)")
+            }
+            return value
+        }
+        set {
+            fatalError("Child relation is get-only.")
+        }
+    }
+
+    public var projectedValue: ChildProperty<From, To> {
+        return self
+    }
+    
+    public var fromId: From.IDValue? {
+        return self.idValue
+    }
+
+    public func query(on database: Database) -> QueryBuilder<To> {
+        guard let id = self.idValue else {
+            fatalError("Cannot query Child relation from unsaved model.")
+        }
+        let builder = To.query(on: database)
+        switch self.parentKey {
+        case .optional(let optional):
+            builder.filter(optional.appending(path: \.$id) == id)
+        case .required(let required):
+            builder.filter(required.appending(path: \.$id) == id)
+        }
+        return builder
+    }
+
+    public func create(_ to: To, on database: Database) -> EventLoopFuture<Void> {
+        guard let id = self.idValue else {
+            fatalError("Cannot save child to unsaved model.")
+        }
+        switch self.parentKey {
+        case .required(let keyPath):
+            to[keyPath: keyPath].id = id
+        case .optional(let keyPath):
+            to[keyPath: keyPath].id = id
+        }
+        return to.create(on: database)
+    }
+}
+
+extension ChildProperty: CustomStringConvertible {
+    public var description: String {
+        self.name
+    }
+}
+
+// MARK: Property
+
+extension ChildProperty: AnyProperty { }
+
+extension ChildProperty: Property {
+    public typealias Model = From
+    public typealias Value = To
+}
+
+// MARK: Database
+
+extension ChildProperty: AnyDatabaseProperty {
+    public var keys: [FieldKey] {
+        []
+    }
+
+    public func input(to input: DatabaseInput) {
+        // Child never has input
+    }
+
+    public func output(from output: DatabaseOutput) throws {
+        let key = From()._$id.field.key
+        if output.contains(key) {
+            self.idValue = try output.decode(key, as: From.IDValue.self)
+        }
+    }
+}
+
+// MARK: Codable
+
+extension ChildProperty: AnyCodableProperty {
+    public func encode(to encoder: Encoder) throws {
+        if let rows = self.value {
+            var container = encoder.singleValueContainer()
+            try container.encode(rows)
+        }
+    }
+
+    public func decode(from decoder: Decoder) throws {
+        // don't decode
+    }
+}
+
+// MARK: Relation
+
+extension ChildProperty: Relation {
+    public var name: String {
+        "Child<\(From.self), \(To.self)>(for: \(self.parentKey))"
+    }
+
+    public func load(on database: Database) -> EventLoopFuture<Void> {
+        self.query(on: database).all().map {
```

Comment:
Why not just do `.first()` here instead?

Reply:
OK, I'll fix this.